core/ops.rs

```diff
@@ -10,30 +10,18 @@ pub type OpId = u32;
 
 pub type Buf = Box<[u8]>;
 
-pub type OpAsyncFuture<E> = Pin<Box<dyn Future<Output = Result<Buf, E>>>>;
+pub type OpAsyncFuture = Pin<Box<dyn Future<Output = Buf>>>;
 
-pub(crate) type PendingOpFuture =
-  Pin<Box<dyn Future<Output = Result<(OpId, Buf), CoreError>>>>;
-
-pub type OpResult<E> = Result<Op<E>, E>;
-
-// TODO(ry) Op::Async should be Op::Async(Pin<Box<dyn Future<Output = Buf>>>)
-// The error should be encoded in the Buf. Notice how Sync ops do not return a
-// result. The Sync and Async should be symmetrical!
-pub enum Op<E> {
+pub enum Op {
   Sync(Buf),
-  Async(OpAsyncFuture<E>),
+  Async(OpAsyncFuture),
   /// AsyncUnref is the variation of Async, which doesn't block the program
   /// exiting.
-  AsyncUnref(OpAsyncFuture<E>),
+  AsyncUnref(OpAsyncFuture),
 }
```

Comment:
The op definition has become completely confused. The idea with ops is that we're passing data pointers back and forth between JS and Rust. Not data pointers and Error objects. There is not any concept of errors in the op definition. Various instantiations of ops may introduce a way to encode errors into these buffers - as JSON dispatch does.